dev/routing/users.js

```diff
@@ -39,11 +40,13 @@ module.exports = function (server) {
       },
       handler: function (request, reply) {
         let user = request.payload;
+        let serverUrl = request.headers.referer;
+        let userEmail = request.payload.email;
         DAL.roles.getByName(Utils.rolesNames.driver, function (err, role) {
           user.roles = [];
           user.roles.push(role._id);
           DAL.users.createUser(request.payload, function (err, docs) {
-            !err ? reply(docs) : reply(JSON.stringify(err));
+            !err ? reply(resetPasswordService.resetPassword(userEmail, serverUrl)) : reply(JSON.stringify(err));
```

Comment:
resetPasswordService.resetPassword returns promise. Why do you try to send promise to frontend?